crates/storage/provider/src/providers/consistent.rs

```diff
@@ -705,7 +705,7 @@ impl<N: ProviderNodeTypes> HeaderProvider for ConsistentProvider<N> {
         self.get_in_memory_or_storage_by_block_range_while(
             range,
             |db_provider, range, _| db_provider.sealed_headers_range(range),
-            |block_state, _| Some(block_state.block_ref().block().header.clone()),
+            |block_state, _| Some(block_state.block_ref().block().header().clone()),
```

Comment:
it looks like the deref coercion causes this to call the `SealedHeader::header` method, which returns an unsealed header. To fix this we could add a method called `header_ref` to our `SealedBlock` type, and then clone.